components/popover_list_members/index.js

```diff
@@ -19,13 +21,16 @@ function makeMapStateToProps() {
     return function mapStateToProps(state, ownProps) {
         const stats = getAllChannelStats(state)[ownProps.channel.id] || {};
         const users = doGetProfilesInChannel(state, ownProps.channel.id, true);
+        const statuses = getUserStatuses(state);
 
         return {
             currentUserId: getCurrentUserId(state),
             memberCount: stats.member_count,
             users,
-            statuses: getUserStatuses(state),
+            statuses,
             teamUrl: getCurrentRelativeTeamUrl(state),
+            manageMembers: canManageMembers(state, ownProps.channel),
+            sortedUsers: sortUsersByStatusAndDisplayName(state, users, statuses),
```

Comment:
We can't call this here unfortunately. `sortUsersByStatusAndDisplayName` returns a new array each time, so it'll cause the `PopoverListMembers` to re-render constantly. Instead, what we can do is make a selector to memoize that for us.

```javascript
const makeGetSortedUsers = (doGetProfilesInChannel) => {
    return createSelector(
        (state, channelId) => doGetProfilesInChannel(state, channelId, true),
        getUserStatuses,
        getTeammateNameDisplay,
        (users, statuses, teammateNameDisplay) => sortUsersByStatusAndDisplayName(users, statuses, teammateNameDisplay)
    );
};
```
This also needs some more changes to `sortUsersByStatusAndDisplayName`, but I'll detail those below